bot/src/api/spotify.rs

```diff
@@ -49,9 +50,22 @@ impl Spotify {
         RequestBuilder::new(self.client.clone(), method, url).token(self.token.clone())
     }
 
+    /// Get user info.
+    async fn me(&self) -> Result<PrivateUser, Error> {
+        let req: RequestBuilder = self.request(Method::GET, &["me"]);
+
+        req.execute().await?.json()
+    }
+
     /// Get my playlists.
     pub async fn playlist(&self, id: String) -> Result<FullPlaylist, Error> {
-        let req = self.request(Method::GET, &["playlists", id.as_str()]);
+        // TODO: cache this value
+        let streamer: PrivateUser = self.me().await.unwrap();
+        let country = streamer.country.unwrap();
+
+        let req = self
+            .request(Method::GET, &["playlists", id.as_str()])
+            .query_param("market", &country);
```

Comment:
Could you pass this in as a parameter to `track` instead. Something like: `market: Option<String>` and `optional_query_param`? Caching can then also be dealt with externally and just worry itself with providing the appropriate parameter. It's also good to maintain that one function call is one request in all API clients, since that's an assumption made in all clients so far.